### ISSUE TITLE
Connectathon 1/2024: Clinical genomics collab

### ISSUE THREAD

**Issue by joeflack4** [HOT-Ecosystem/hapi-fhir-jpaserver-starter#136]
## Overview
Melissa exposited a use case that this group has had for a while, which is “Give me on all the lab terms for a given gene”. However, it may be that we don't have the appropriate gene terms for this. We could do something similar for chemistry terms, perhaps.

Joe has more notes on this in his personal notes and will need to follow up with Shahim for clarification on potential next steps.

## Use case 2: Provide HGNC Code Systems
<details><summary>Details</summary>
<p>

These are defined here: https://build.fhir.org/ig/HL7/genomics-reporting/ValueSet-hgnc-vs.html
1. Include all codes defined in http://www.genenames.org
2. Include all codes defined in http://www.genenames.org/genegroup

There is a link to a THO page with CodeSystem URIs, sow e'd have to use those.

We need to investigate their content. 

</p>
</details> 

### Sub-task list
If (in step 2) they tell us that they don't like HGNC, step 2-3 are required. But we could still skip 2-3 regardless and show them the HGNC proof of concept.
- [ ] 1. **Spreadsheet 1: Current mappings**
  - Who/by when:
    - Send current mappings @ShahimEssaid @by-Fri-11/17
    - Review current mappings @Melissa/@group @by-wed-12/01
  - Provide a CSV to Melissa that has all currently mapped and unmapped LOINC gene codes. See what is needed next. Perhaps we will want to share with clinical genomics group too?
  - Possibilities:  (a) not happy with what LOINC is providing, (b) many LOINC parts not mapped yet to HGNC
- [ ] 2. **Spreadsheet 2: New mappings**
  - Who: Likely @ShahimEssaid but @joeflack4 could also do
  - By when: @by-wed-12/06
  - Depends on: We won't be able to start this until we hear back from people from step (1). And we may not get that back by Friday after Thanksgiving, @wed-12/01
  - Map to HGNC if they're happy with that, or other ontology based on their desire
  - What tool  to use for to create mappings (based on Name Entity Recognition). Ask who? Melissa?
- [ ] 3. **Spreadsheet 3: Curated**
  - Who: **@curators-who?**
  - By when: @by-fri-12/15~
- [ ] 4. **SSSOM**?
  - Who: @ShahimEssaid  or probably @joeflack4 
  - By when: @by-wed1-2/20
  - Normalize spreadsheet from previous step
- [ ] 5. **OWL**
  - Who: @ShahimEssaid 
  - By when: @by-fri1/05
  - How: spreadsheet/SSSOM -> CompLoinc -> OWL
- [ ] 6. **End products**
  - [ ] 6.1. **FHIR**
    - Who: @joeflack4 
    - By when: @by-1/16-connectathon 
    - Some means of answering question "here is a gene code. what are the LOINC lab terms associated"
    - Code system w/ properties? Or ConceptMap? both?
  - [ ] 6.2. **Custom tool?**
    - Who: @joeflack4 
    - By when: @by-1/16-connectathon if possible
    - Could be a simple UI backed by a Python server running OAK

#### 1. Spreadsheet 1: Current mappings
As part of step (1) in "main tasks", also ask Melissa / Clinical Genomics group this question:
  - Melissa said that this use case would be useful to Clinical genomics group: "Given gene, what are associated labs?" Has she heard of that being useful to that group? If so, does she have a notion of how they would like they like to do something like that? As a spreadsheet? A web UI? As specifically part of FHIR that they would query?
  - Who is this for? Is this thought of as a TIMS objective service for Clinical Genomics group? Or is this something that primarily benefits CompLOINC? Or something else / don't know yet?
  - Should the clinical genomics working group also review mappings or just us?
  - What tool to use for to create mappings (based on Name Entity Recognition)

### Other tasks
- [ ] 1. Could discuss with clinical genomics working group at one of their meetings. Virtual and happen on the 29th. 

### Additional info
- LOINC gene mappings as of 2023/11/16: [part-to-gene.csv](https://github.com/HOT-Ecosystem/issues/files/13384519/part-to-gene.csv)

**Comment by joeflack4** [comment]
Shahim found several existing solutions, so we're wondering if we really need to do anything here. For example:
1. Athena: e.g. on this [gene page](https://athena.ohdsi.org/search-terms/terms/40795325), you can click the top link on the right to open see this [lab test](https://athena.ohdsi.org/search-terms/terms/42529183).
2. UMLS: e.g. on this [gene page](https://uts.nlm.nih.gov/uts/umls/concept/C1335212), you click the [LOINC link](https://uts.nlm.nih.gov/uts/umls/vocabulary/LNC/LP101031-5) at bottom and you can find several labs listed there, e.g. [this one](https://uts.nlm.nih.gov/uts/umls/vocabulary/LNC/85507-2).
3. LOINC: There are some CSVs you can download, and after doing a join you can get all the gene to LOINC lab mappings.

For (3), if you download the LOINC release, there are some relevant instructions in `AccessoryFiles/PartFile/PartFileReadMe.txt`, specifically the part starting at line 71 explaining the `LoincPartLink_Supplementary.csv`:
<details><summary>Details</summary>
<p>

LoincPartLink_Supplementary.csv

Overview:
This file contains every LOINC term included in the current LOINC release and their part link types other than Primary, Radiology, and Document Ontology. Link types in this file include:  
1. "Search" - parts that are only linked to a given term in order to facilitate efficient searching and location of that term (note that parts that have a "Search" type link to one LOINC term may have a "Primary" link to another LOINC term); 
2. "DetailedModel" - the parts that comprise all of the more detailed compositional subparts that create a fully defined term according to the LOINC semantic model;
3. "SyntaxEnhancement" - the parts linked to a LOINC term that represent useful pieces of the LOINC term name that are not officially part of the semantic model;
4. "SemanticEnhancement" - the parts linked to a LOINC term that are part of the LOINC term name but not officially part of the semantic model and are used to link terms to external sources or coding systems; and
5. "Metadata" - parts that define accessory term attributes that are coded but not part of the LOINC term name (e.g. CLASS).

Note that Document Ontology and Radiology terms will not have "DetailedModel" links because the general detailed model is superseded by the Document Ontology and Radiology models.

Columns:
 - LoincNumber - the LOINC term identifier, which is a unique numeric identifier with a check digit
 - LongCommonName - the LOINC Long Common Name
 - PartNumber - the LOINC part number, which is a unique identifier that starts with the prefix "LP". This field is also contained in the Part and PartRelatedCodeMapping files and defines the relationship between the three tables contained in the LOINC Part file.
 - PartName - the name of the Part. Examples: hemoglobin, clavicle
 - PartCodeSystem - the coding system to which the Part belongs
 - PartTypeName - the type of Part. Examples: Component, Method
 - LinkTypeName - the type of link between the LOINC term and the Part in a given row. Values include "Search", "DetailedModel", "DetailedModel", "SemanticEnhancement", and "Metadata" as described above
 - Property - the attribute that the Part represents

Sort order:
This file is sorted by alphabetically by the LoincNumber; then by the LinkTypeName in the following order (note that not all terms will have all of these links): "DetailedModel", "SyntaxEnhancement", "SemanticEnhancement", "Search", and "Metadata"; and finally by the PartNumber.

</p>
</details> 

As Shahim said [on Slack](https://tims-4sl3026.slack.com/archives/D031F9E35QQ/p1700183333895409):
> any row in that file that has `http://loinc.org/property/analyte-gene` in the `Property` column is a link between a LOINC code (col 1) and the loinc part that represents a gene (col 3).
> so, a join between the above csv and the one from my email (by loinc part id) will be a mapping from LOINCs to HGNC for example

---

Some more related information about these gene relationships, e.g. `analyte-gene`, and the supporting data model:
- https://loinc.org/kb/enriched-linkages-between-loinc-terms-and-loinc-parts/#semantic-enhancement
- https://loinc.org/kb/enriched-linkages-between-loinc-terms-and-loinc-parts/#examples-3

**Comment by joeflack4** [comment]
@Leskena2 @ShahimEssaid Dang, we thought through this issue so well. But unfortunately nothing ended up happening with this. Maybe we can reopen some variation of this in the future. Closing.
